test/cql-pytest/test_describe.py

```diff
@@ -552,6 +552,50 @@ def test_whitespaces_in_table_options(cql, test_keyspace, scylla_only):
         desc = "\n".join([d.create_statement for d in cql.execute(f"DESC TABLE {tbl}")])
         assert re.search(regex, desc) == None
 
+# Randomly create many UDTs, with dependencies between them 
+# and validate if describe displays them in correct order.
+# UDTs should be sorted topologically, meaning if UDT `a`
+# is used to create UDT `b`, then `a` should be before `b`.
+def test_udt_sorting(scylla_only, cql, test_keyspace, random_seed):
+    repeat = 10 # test it randomized, so repeat it inside the test to increase coverage
+    max_selected_udts = 5 # max numer of UDTs that can be used to create a new UDT
+
+    for _ in range(repeat):
+        created_udts = []
+        udts_cnt = random.randint(10, 50)
+
+        with ExitStack() as stack:
+            # Create `udts_cnt` many UDTs
+            while len(created_udts) < udts_cnt:
+                select_cnt = random.randint(0, min(max_selected_udts, len(created_udts))) # how many UDTs we want to use to create a new one
+                udt_body = "(a int)"
+                if select_cnt != 0:
+                    selected_udts = random.sample(created_udts, select_cnt) # randomly select `select_cnt` already created UDTs
+                    udt_body = create_udt_body(selected_udts)
+                
+                udt = stack.enter_context(new_type(cql, test_keyspace, udt_body))
+                created_udts.append(udt)
+            
+            # Describe all UDTs, iterate over all of them and verify the order is correct .
+            # (If UDT `u` is used create UDT `w`, `u` has to be described before `w`)
+            desc = cql.execute(f"DESC KEYSPACE {test_keyspace}").all()
+            visited_udts = set()
+            for row in desc[1:]: # skip keyspace entry
+                # Extract UDTs used in current UDT
+                # Names of all UDTs begin with `unique_name_prefix`
+                for it in re.finditer(f"<({unique_name_prefix}[^\s\>]+)>", row.create_statement):
+                    assert it.group(1) in visited_udts
+                visited_udts.add(row.name)
+            
+            assert len(visited_udts) == len(created_udts)
+
+            # Drop all UDTs and recreate them from the description.
+            # We can safely drop `test_keyspace` here, because it'll be recreated
+            # (first entry of `DESC KEYSPACE ks` is keyspace create statement)
+            cql.execute(f"DROP KEYSPACE {test_keyspace}")
+            for row in desc:
+                cql.execute(row.create_statement)
```

Comment:
I thought test_keyspace gives a unique name each time it's executed?

Reply:
> I thought test_keyspace gives a unique name each time it's executed?

No, test_keyspace is a single (session-scope) fixture when many tests uses. We don't run these tests in parallel, but we do run them sequentially and the fact they reuse the same keyspace shaves a few milliseconds from each test. Moreover, multiple tests can even reuse the same table. In some cases we could even build a single table with content and have it read by multiple tests (I do this heavily in test/alternator but not in cql-pytest).

To cut a long story short, a test that uses the shared keyspace shouldn't modify it or delete it. If a test wants a new temporary keyspace it can easily do this (there's a new_keyspace context manager). But in this specific test I think even this is unrelated to the goal of this test.